test/AutoRest.TestServerLowLevel.Tests/dpg-customization.cs

```diff
@@ -93,68 +93,344 @@ await foreach (Page<Product> page in allPages.AsPages())
         });
 
         [Test]
-        public Task RawLRO() => Test(async (host) =>
+        [TestCase(true)]
+        [TestCase(false)]
+        public Task RawLRO(bool nestedScopeSuppression) => Test(async (host) =>
         {
             using var diagnosticListener = new ClientDiagnosticListener("dpg_customization_LowLevel", asyncLocal: true);
             CollectionAssert.IsEmpty(diagnosticListener.Scopes);
 
-            Operation<BinaryData> result = await new DPGClient(Key, host, null).LroAsync(WaitUntil.Started, "raw");
+            Operation<BinaryData> result = await new DPGClient(Key, host, null, nestedScopeSuppression).LroAsync(WaitUntil.Started, "raw");
             diagnosticListener.AssertAndRemoveScope("DPGClient.Lro");
             CollectionAssert.IsEmpty(diagnosticListener.Scopes);
 
             await result.WaitForCompletionAsync();
             diagnosticListener.AssertAndRemoveScope("DPGClient.Lro.WaitForCompletion");
-            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+            if (nestedScopeSuppression)
+            {
+                CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+            }
 
             JsonData responseBody = JsonData.FromBytes(result.Value.ToMemory());
             Assert.AreEqual("raw", (string)responseBody["received"]);
         });
 
         [Test]
-        public Task HandwrittenModelLro() => Test(async (host) =>
+        [TestCase(true)]
+        [TestCase(false)]
+        public Task RawLROResponseOperationOfT(bool nestedScopeSuppression) => Test(async (host) =>
+        {
+            using var diagnosticListener = new ClientDiagnosticListener("dpg_customization_LowLevel", asyncLocal: true);
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
+            Operation<BinaryData> result = await new DPGClient(Key, host, null, nestedScopeSuppression).LroAsync(WaitUntil.Started, "raw");
+            diagnosticListener.AssertAndRemoveScope("DPGClient.Lro");
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
+            await result.WaitForCompletionResponseAsync();
+            diagnosticListener.AssertAndRemoveScope("DPGClient.Lro.WaitForCompletion");
```

Comment:
Could you confirm with arch it should be `DPGClient.Lro.WaitForCompletion` or `DPGClient.Lro.WaitForCompletionResponse`?

Reply:
My understanding is that the scope should have the same name as the public API call, per https://github.com/Azure/autorest.csharp/issues/2233#issuecomment-1127819652.  It would be worth understanding why this isn't happening here to learn whether it is a bug or there is a reason to make an exception in this case.